src/components/Modal/ManageTaskModal.tsx

```diff
@@ -11,45 +11,106 @@ import { ReactComponent as AssigneeIcon } from '@assets/assignee.svg';
 import { ReactComponent as CalendarIcon } from '@assets/calendar.svg';
 import { CreateTask } from '@models/task.model';
 import { CustomListbox } from '@components/Listbox';
-import { useMutation, useQuery } from '@apollo/client';
+import { gql, useMutation, useQuery } from '@apollo/client';
 import { GET_ASSIGNEES } from '@gql/queries.graph';
 import { Loader } from '@components/Common';
 import { User } from '@models/user.model';
 import Avatar from '@components/Common/Avatar';
 import { toast } from 'react-hot-toast';
 import { CREATE_TASK, UPDATE_TASK } from '@gql/mutations.graph';
-import { AppContenxt } from '@ctx/app.ctx';
 import 'react-datepicker/dist/react-datepicker.css';
 import { toDigit } from '@utils/task';
-import { formatTag } from '@utils/tags';
-import { Tag } from '@models/common.model';
+import { TaskContext } from '@ctx/task.ctx';
+import { Reference } from '@apollo/client/cache';
 import { ManageTaskModalProps } from './props';
 import Modal from './Modal';
 
 const ManageTaskModal: FC<ManageTaskModalProps> = (
     {
-        modalStatus, openModal, closeModal, task, mode,
+        modalStatus, openModal, closeModal,
     },
 ) => {
-    const { tasks } = useContext(AppContenxt)!;
+    const { editableTask: task, setEditableTask } = useContext(TaskContext)!;
     const [dueDate, setDueDate] = useState<Date>();
     const { data, loading } = useQuery(GET_ASSIGNEES);
-    const [mutateFn, { loading: mutateLoading }] = useMutation(CREATE_TASK);
-    const [mutateUpdateFn, { loading: mutateUpdating }] = useMutation(UPDATE_TASK);
+
+    const [createTask, { loading: mutateLoading }] = useMutation(CREATE_TASK, {
+        update(cache, { data: { createTask: createResult } }) {
+            cache.modify({
+                fields: {
+                    tasks(existingTasks, { readField, storeFieldName }) {
+                        if (!storeFieldName.includes('TODO')) return existingTasks;
+
+                        const newTaskRef = cache.writeFragment({
+                            data: createResult,
+                            fragment: gql`
+                                fragment NewTask on Task {
+                                    id
+                                    name
+                                    dueDate
+                                    pointEstimate
+                                    tags
+                                    position
+                                    assignee {
+                                        avatar
+                                        fullName
+                                    }
+                                }
+                            `,
+                        })!;
+
+                        if (existingTasks.some((ref: Reference) => readField('id', ref) === createResult.id)) return existingTasks;
+
+                        return [...existingTasks, newTaskRef];
+                    },
+                },
+            });
+        },
+    });
+
```

Comment:
Updating task in cache from the new values get from the response, and `storeFieldName`